### ISSUE TITLE
Incorrect Native Coin (BNB) Conversion in GetInternalTransactionByAddress

### ISSUE THREAD

**Issue by migregorio** [dannyhu/AtomicCore#8]
The amount of BNB displayed for internal transaction results ("TxValue" field) is different from when I call the Bscscan API directly ("value" field).

Example here:
https://api.bscscan.com/api?module=account&action=txlistinternal&address=0x091dD81C8B9347b30f1A4d5a88F92d6F2A42b059&startblock=0&endblock=latest&sort=desc&apikey=YourApiKeyToken

Transaction 0x66934ec4c4c02df9b9b018fd37510a967c34f678510ff7ea0426d2d248474da5 shows 0.241015459589588883 using this library but shows 358424896370793 when using the API directly. Something about the conversion isn't correct. Maybe it's converting to ETH instead of BNB.

**Comment by dannyhu** [comment]
I will check the code later




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dannyhu/AtomicCore"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月23日(星期一) 凌晨1:59
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[dannyhu/AtomicCore] Incorrect Native Coin (BNB) Conversion in GetInternalTransactionByAddress (Issue #8)





 
The amount of BNB displayed for internal transaction results ("TxValue" field) is different from when I call the Bscscan API directly ("value" field).
 
Example here:
 https://api.bscscan.com/api?module=account&amp;action=txlistinternal&amp;address=0x091dD81C8B9347b30f1A4d5a88F92d6F2A42b059&amp;startblock=0&amp;endblock=latest&amp;sort=desc&amp;apikey=YourApiKeyToken
 
Transaction 0x66934ec4c4c02df9b9b018fd37510a967c34f678510ff7ea0426d2d248474da5 shows 0.241015459589588883 using this library but shows 358424896370793 when using the API directly. Something about the conversion isn't correct. Maybe it's converting to ETH instead of BNB.

**Comment by dannyhu** [comment]
Yes, I found the problem, I treated the decimal as hexadecimal data to cause data errors, thank you for finding the bug for me
Please update to V1.08
thank you very much


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "dannyhu/AtomicCore"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月23日(星期一) 凌晨1:59
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[dannyhu/AtomicCore] Incorrect Native Coin (BNB) Conversion in GetInternalTransactionByAddress (Issue #8)





 
The amount of BNB displayed for internal transaction results ("TxValue" field) is different from when I call the Bscscan API directly ("value" field).
 
Example here:
 https://api.bscscan.com/api?module=account&amp;action=txlistinternal&amp;address=0x091dD81C8B9347b30f1A4d5a88F92d6F2A42b059&amp;startblock=0&amp;endblock=latest&amp;sort=desc&amp;apikey=YourApiKeyToken
 
Transaction 0x66934ec4c4c02df9b9b018fd37510a967c34f678510ff7ea0426d2d248474da5 shows 0.241015459589588883 using this library but shows 358424896370793 when using the API directly. Something about the conversion isn't correct. Maybe it's converting to ETH instead of BNB.
